### PR TITLE
fix: unexpected panic in some cases

### LINKED ISSUE #17: unexpected panic in some cases
need more investigation, seems to be when "trying" to parse the response body, maybe a dropped connection or something?

fyi, I was querying the API of CircleCI

```
panic: runtime error: slice bounds out of range [:5302] with capacity 4096 [recovered]
        panic: runtime error: slice bounds out of range [:5302] with capacity 4096

goroutine 92 [running]:
io/ioutil.readAll.func1(0xc005ed98f0)
        /home/moul/.gvm/gos/go1.13.6/src/io/ioutil/ioutil.go:30 +0x101
panic(0xd4aa40, 0xc005f73280)
        /home/moul/.gvm/gos/go1.13.6/src/runtime/panic.go:679 +0x1b2
bufio.(*Reader).Read(0xc005efdf20, 0xc00054b6b8, 0x4, 0x4, 0xc005ed9570, 0x40bec5, 0xcb9700)
        /home/moul/.gvm/gos/go1.13.6/src/bufio/bufio.go:237 +0x3f0
io.ReadAtLeast(0xeeb780, 0xc005efdf20, 0xc00054b6b8, 0x4, 0x4, 0x4, 0xc005ed95f8, 0x6796c4, 0xc0e4d4ff23)
        /home/moul/.gvm/gos/go1.13.6/src/io/io.go:310 +0x87
io.ReadFull(...)
        /home/moul/.gvm/gos/go1.13.6/src/io/io.go:329
compress/flate.(*decompressor).dataBlock(0xc00054a600)
        /home/moul/.gvm/gos/go1.13.6/src/compress/flate/inflate.go:629 +0x9a
compress/flate.(*decompressor).nextBlock(0xc00054a600)
        /home/moul/.gvm/gos/go1.13.6/src/compress/flate/inflate.go:314 +0x203
compress/flate.(*decompressor).Read(0xc00054a600, 0xc006017816, 0x5ea, 0x5ea, 0x2000, 0xd1695a8c, 0x0)
        /home/moul/.gvm/gos/go1.13.6/src/compress/flate/inflate.go:347 +0x77
compress/gzip.(*Reader).Read(0xc005fa2580, 0xc006017816, 0x5ea, 0x5ea, 0x1a16, 0x0, 0x0)
        /home/moul/.gvm/gos/go1.13.6/src/compress/gzip/gunzip.go:251 +0x87
net/http.(*gzipReader).Read(0xc005f755a0, 0xc006017816, 0x5ea, 0x5ea, 0x1a16, 0x0, 0x0)
        /home/moul/.gvm/gos/go1.13.6/src/net/http/transport.go:2640 +0xcd
net/http.(*cancelTimerBody).Read(0xc005f755c0, 0xc006017816, 0x5ea, 0x5ea, 0x1a16, 0x0, 0x0)
        /home/moul/.gvm/gos/go1.13.6/src/net/http/client.go:863 +0x5c
bytes.(*Buffer).ReadFrom(0xc005ed9888, 0x7f55a413a2a8, 0xc005f755c0, 0x40ad7b, 0xc000020000, 0xcb9700)
        /home/moul/.gvm/gos/go1.13.6/src/bytes/buffer.go:204 +0xb4
io/ioutil.readAll(0x7f55a413a2a8, 0xc005f755c0, 0x200, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/moul/.gvm/gos/go1.13.6/src/io/ioutil/ioutil.go:36 +0x100
io/ioutil.ReadAll(...)
        /home/moul/.gvm/gos/go1.13.6/src/io/ioutil/ioutil.go:45
github.com/Bearer/bearer-go.newRecord(0xc000106400, 0xc005f14240, 0xbf8d1063d894cd2c, 0x2112a024, 0x14dfce0, 0xbf8d106411f91be0, 0x5611b909, 0x14dfce0, 0x0, 0x0, ...)
        /home/moul/go/src/github.com/Bearer/bearer-go/agent.go:124 +0x6e4
github.com/Bearer/bearer-go.(*Agent).RoundTrip.func1(0xc000106400, 0xc005f14240, 0xbf8d1063d894cd2c, 0x2112a024, 0x14dfce0, 0xbf8d106411f91be0, 0x5611b909, 0x14dfce0, 0x0, 0x0, ...)
        /home/moul/go/src/github.com/Bearer/bearer-go/agent.go:92 +0x12f
created by github.com/Bearer/bearer-go.(*Agent).RoundTrip
        /home/moul/go/src/github.com/Bearer/bearer-go/agent.go:91 +0x2c6
```

### PR DESCRIPTION
**Pull request by moul**
* [x] fixes #17
* [x] add circleci example (used for testing)
* [x] configure .gitattributes for better PR review experience

### CODE REVIEW

File: agent.go [hunk 1]
```diff
@@ -120,8 +125,11 @@ func newRecord(req *http.Request, resp *http.Response, start, end time.Time, req
 		RequestHeaders:  goHeadersToBearerHeaders(req.Header),
 		ResponseHeaders: goHeadersToBearerHeaders(resp.Header),
 	}
-	if resp.Body != nil && isParseableContentType.MatchString(record.RequestContentType()) {
-		buf, _ := ioutil.ReadAll(resp.Body)
+	if roundtripError == nil && resp.Body != nil && isParseableContentType.MatchString(record.RequestContentType()) {
+		fmt.Println("AAA", resp.StatusCode, resp.ContentLength, resp.Header)
```
Comment (cfabianski) [inline agent.go]: remove?

File: agent.go [hunk 2]
```diff
@@ -120,8 +125,11 @@ func newRecord(req *http.Request, resp *http.Response, start, end time.Time, req
...
+		buf, err := ioutil.ReadAll(resp.Body)
+		fmt.Println("BBB", len(buf), err)
```
Comment (cfabianski) [inline agent.go]: Remove?

File: agent.go [hunk 3]
```diff
@@ -120,8 +125,11 @@ func newRecord(req *http.Request, resp *http.Response, start, end time.Time, req
...
+		fmt.Println("CCC", string(buf))
```
Comment (cfabianski) [inline agent.go]: Remove?

File: agent.go (hunk 1 above)
Comment (moul) [reply agent.go]: It's still WIP (see title)

**Pull request merged**